.github/workflows/attach-static-libs.yaml

```diff
@@ -0,0 +1,170 @@
+      matrix:
+        include:
+          - os: ubuntu-latest
+            target: x86_64-unknown-linux-gnu
+          - os: ubuntu-22.04-arm
+            target: aarch64-unknown-linux-gnu
+          - os: macos-latest
+            target: aarch64-apple-darwin
+          - os: macos-13
+            target: x86_64-apple-darwin
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: arduino/setup-protoc@v3
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - uses: Swatinem/rust-cache@v2
+
+      - name: Build for ${{ matrix.target }}
+      # TODO: add ethhash feature flag after updating FFI tests
+        run: cargo build --profile maxperf --features logger --target ${{ matrix.target }} -p firewood-ffi
+
+      - name: List compiled target directory
+        run: find target -type f | sort
+        
+      - name: Upload binary
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ matrix.target }}
+          path: target/${{ matrix.target }}/maxperf/libfirewood_ffi.a
+          if-no-files-found: error
+
+  # Collect all the static libraries built on the previous matrix of jobs
+  # and add them into ffi/libs directory.
+  # We commit and push this as a new branch with "--force" to overwrite
+  # the previous static libs that will not be on our branch.
+  push-firewood-ffi-libs:
+    needs: build-firewood-ffi-libs
+    runs-on: ubuntu-latest
+    outputs:
+      target_branch: ${{ steps.determine_branch.outputs.target_branch }}
+    steps:
+      - name: Determine branch name
+        id: determine_branch
+        run: |
+          if [[ "${{ github.event_name }}" == "workflow_dispatch" && -n "${{ github.event.inputs.create_branch_name }}" ]]; then
+            export target_branch="${{ github.event.inputs.create_branch_name }}-ffi-libs"
+            echo "Using workflow input + -ffi-libs as target branch: $target_branch"
+            echo "target_branch=$target_branch" >> "$GITHUB_OUTPUT"
+          elif [[ "${{ github.event_name }}" == "push" ]]; then
+            export target_branch="${GITHUB_REF#refs/heads/}-ffi-libs"
+            echo "Using tag/branch name + -ffi-libs as target_branch: $target_branch"
+            echo "target_branch=$target_branch" >> "$GITHUB_OUTPUT"
+          elif [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            export target_branch="${{ github.event.pull_request.head.ref }}-ffi-libs"
+            echo "Using PR head name + -ffi-libs as target branch: $target_branch"
+            echo "target_branch=$target_branch" >> "$GITHUB_OUTPUT"
+          else
+            echo "No valid input or tag found."
+            exit 1
+          fi
+
+      - uses: actions/checkout@v4
+        with:
+          path: firewood
+
+      - uses: actions/checkout@v4
+        with:
+          repository: aaronbuchwald/firewood-go
+          token: ${{ secrets.FORK_FIREWOOD_GO_GITHUB_TOKEN }}
+          path: firewood-go
```

Comment:
Still waiting for approval on a PAT scoped to ava-labs/firewood-go, so I migrated to my own fork so I can create my own scoped token for now.